python/pyarrow/src/arrow/python/pyarrow_api.h

```diff
@@ -16,4 +16,4 @@
 // under the License.
 
 // For backward compatibility.
-#include "arrow/python/lib_api.h"
+#include "lib_api.h"
```

Comment:
The modifications in this header are tightly bound to the `cython_generated_headers` target being generated in the Meson configuration. I don't think Cython gives you any control over the output directory / filename of cythonized modules, so Meson just places them in the build directory next to the module used to create the. 

PyArrow is expecting these files to be placed in `arrow/python`, so I am wondering if I need to add another custom_target that moves the files around in the build directory after they are generated, or if there's yet a better approach